src/extension/app/store/app.js

```diff
@@ -532,7 +532,22 @@ export class AppStore {
    */
   getContentSourceLabel() {
     const { preview } = this.status;
-    return preview?.sourceLocation.includes('onedrive:') ? 'SharePoint' : 'Google Drive';
+    const { sourceLocation } = preview;
+
+    if (sourceLocation) {
+      return sourceLocation.startsWith('onedrive:')
+        ? 'SharePoint'
+        : sourceLocation.startsWith('gdrive:')
+          ? 'Google Drive'
+          : 'BYOM';
+    }
+
+    const { mountpoint } = this.siteStore;
+    return mountpoint.includes('.google.com')
+      ? 'Google Drive'
+      : mountpoint.includes('.sharepoint.com') || mountpoint.includes('/Shared%20Documents/sites/')
```

Comment:
I've had a similar line flagged in a security review as insufficient URL check, because it could match anywhere in the URL... it's safer to use the `URL` object here:
```suggestion
    const { host, pathname } = new URL(mountpoint);
    return host.endsWith('.google.com')
      ? 'Google Drive'
      : host.endsWith('.sharepoint.com') || pathname.includes('/Shared%20Documents/sites/')
```

Reply:
Ah, yeah that makes sense. I think the damage that could be done here is just changing a string in the UI but point taken. Maybe we will use this function for something else in the future.